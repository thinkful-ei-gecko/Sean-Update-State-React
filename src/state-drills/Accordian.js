import React from 'react'

class Accordian extends React.Component{

    static defaultProps={
        text:[],
    };

    state={
        clicked:false,
        index:-1,
    }
    
    clickHandler=(event)=>{
        if(event.target.value){
            this.setState({clicked:!this.state.clicked,
                           index:event.target.value})
        }
    }

    render(){

        return(
            <ul className='Accordian'>
                {this.props.text.map((section, index) => {
                    return(
                    <li key={index}>
                        <button type='button' onClick={(event)=>this.clickHandler} value={index}>Click Me</button>
                        <p index={index}>{this.state.clicked && index == this.state.index ? section.content: null}</p>
                    </li>
                    )
                })}
            </ul>
        )
    }
}

export default Accordian