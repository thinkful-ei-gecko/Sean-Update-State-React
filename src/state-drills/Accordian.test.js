import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Accordian from './Accordian';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]


describe('Accordian Component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders and empty li when the sections prop is not supplied', () => {
    const wrapper = shallow (<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders no sections as active by default', () => {
    const wrapper = shallow (<Accordian text={sectionsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('opens a clicked section', () => {
    const wrapper = shallow(<Accordian text={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordian text={sectionsProp} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})