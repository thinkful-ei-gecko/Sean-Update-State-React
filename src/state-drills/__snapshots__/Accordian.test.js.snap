// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only opens the last section when multiple sections have been clicked 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
      value={0}
    >
      Click Me
    </button>
    <p
      index={0}
    />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
      value={1}
    >
      Click Me
    </button>
    <p
      index={1}
    />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
      value={2}
    >
      Click Me
    </button>
    <p
      index={2}
    />
  </li>
</ul>
`;

exports[`Accordian Component opens a clicked section 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
      value={0}
    >
      Click Me
    </button>
    <p
      index={0}
    />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
      value={1}
    >
      Click Me
    </button>
    <p
      index={1}
    />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
      value={2}
    >
      Click Me
    </button>
    <p
      index={2}
    />
  </li>
</ul>
`;

exports[`Accordian Component renders and empty li when the sections prop is not supplied 1`] = `
<ul
  className="Accordian"
/>
`;

exports[`Accordian Component renders no sections as active by default 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
      value={0}
    >
      Click Me
    </button>
    <p
      index={0}
    />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
      value={1}
    >
      Click Me
    </button>
    <p
      index={1}
    />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
      value={2}
    >
      Click Me
    </button>
    <p
      index={2}
    />
  </li>
</ul>
`;
